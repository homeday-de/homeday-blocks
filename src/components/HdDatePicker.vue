<template>
  <date-pick
    v-bind="{...$props, ...$attrs}"
    :value="value"
    v-on="$listeners"
    @input="handleInput"
    :editable="!disabled"
  >
    <slot
      v-bind="{...$props, ...$attrs}"
      v-on="$listeners"
    />
  </date-pick>
</template>

<script>
import DatePick from 'vue-date-pick';

export default {
  name: 'HdDatePicker',
  components: { DatePick },
  props: {
    value: {
      type: String,
      default: '',
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    format: {
      type: String,
      default: 'YYYY.MM.DD',
    },
    nextMonthCaption: {
      type: String,
      default: 'Nächsten Monat',
    },
    prevMonthCaption: {
      type: String,
      default: 'Vorheriger Monat',
    },
    setTimeCaption: {
      type: String,
      default: 'Zeit einstellen',
    },
    weekdays: {
      type: Array,
      default: () => ([
        'Mo', 'Di', 'Mi', 'Do', 'Fr', 'Sa', 'So',
      ]),
    },
    months: {
      type: Array,
      default: () => ([
        'Januar', 'Februar', 'März', 'April',
        'Mai', 'Juni', 'Juli', 'August',
        'September', 'Oktober', 'November', 'Dezember',
      ]),
    },
  },
  methods: {
    handleInput(value) {
      this.$emit('input', value);
    },
  },
};
</script>

<style lang="scss">
$vdpColor: #5BA4FF;
@import 'vue-date-pick/src/vueDatePick.scss';
</style>
