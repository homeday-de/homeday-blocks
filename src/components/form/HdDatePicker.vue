<template>
  <date-pick
    v-bind="{...$props, ...$attrs}"
    :value="value"
    @input="handleInput"
  />
</template>

<script>
import DatePick from 'vue-date-pick';
import 'vue-date-pick/dist/vueDatePick.css';

export default {
  components: { DatePick },
  props: {
    value: {
      type: String,
      default: '',
    },
    displayFormat: {
      type: String,
      default: 'YYYY.MM.DD',
    },
    weekdays: {
      type: Array,
      default: () => ([
        'So', 'Mo', 'Di', 'Mi', 'Do', 'Fr', 'Sa',
      ]),
    },
    months: {
      type: Array,
      default: () => ([
        'Januar', 'Februar', 'MÃ¤rz', 'April',
        'Mai', 'Juni', 'Juli', 'August',
        'September', 'Oktober', 'November', 'Dezember',
      ]),
    },
    startWeekOnSunday: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    handleInput(value) {
      this.$emit('input', value);
    },
  },
};
</script>
