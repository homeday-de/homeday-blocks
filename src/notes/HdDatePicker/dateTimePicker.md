# Datetime picker

You can create a date picker with controls for hours and minutes.

## Code example

```html
<template>
  <HdDatePicker
    v-model="date"
    :pickTime="true"
    :use12HourClock="true"
    :format="'YYYY-MM-DD HH:mm'"
    :displayFormat="'YYYY.MM.DD H:mm A'"
  />
</template>

<script>
import { HdDatePicker } from 'homeday-blocks';

export default {
  components: { HdDatePicker },
  data() {
    return {
      date: '2020.06.01 14:30',
    };
  },
};
</script>
```
