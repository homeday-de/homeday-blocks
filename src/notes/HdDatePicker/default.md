# `<HdDatePicker>`

The `<HdDatePicker>` is a wrapper component over **[vue-date-pick][vue_date_pick]**. We decided to use this library becauseit is a simple library, with active development and a very powerful API solving our current needs for picking dates.

It can handle:

- i18n
- Custom styles
- Different formats
- Cross browser support

## Basic usage

```html
<template>
  <HdDatePicker
    v-model="date"
  />
</template>

<script>
import { HdDatePicker } from 'homeday-blocks';

export default {
  components: { HdDatePicker },
  data() {
    return {
      date: '2020-06-01',
    };
  },
};
</script>
```

<!-- Links references -->

[vue_date_pick]: https://github.com/dbrekalo/vue-date-pick
