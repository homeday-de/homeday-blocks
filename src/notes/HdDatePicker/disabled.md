# `<HdDatePicker>` - disabled

You can disable the input. This could be useful when you want to show some date, but not let the user to edit it.

## Code example

```html
<template>
  <HdDatePicker
    v-model="date"
    :editable="false"
  />
</template>

<script>
import { HdDatePicker } from 'homeday-blocks';

export default {
  components: { HdDatePicker },
  data() {
    return {
      date: '2020-06-01',
    };
  },
};
</script>
```
