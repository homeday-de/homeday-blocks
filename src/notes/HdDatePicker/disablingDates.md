# Disabling dates

You can disable some dates on the `<HdDatePicker>` with the prop `isDateDisabled`. It must be a function that returns a boolean value for a date.

## Code example

```html
<template>
  <HdDatePicker
    v-model="date"
    :isDateDisabled="isFutureDate"
  />
</template>

<script>
import { HdDatePicker } from 'homeday-blocks';

export default {
  components: { HdDatePicker },
  data() {
    return {
      date: '2020-06-01',
    };
  },
  methods: {
    isFutureDate(date) {
      const currentDate = new Date();
      return date > currentDate;
    },
  },
};
</script>
```
