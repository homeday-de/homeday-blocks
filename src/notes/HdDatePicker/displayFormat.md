# `<HdDatePicker>` - display format

Use `displayFormat` prop when UI date format is different from raw data format.

## Code example

```html
<template>
  <HdDatePicker
    v-model="date"
    :displayFormat="MM-YYYY"
  />
</template>

<script>
import { HdDatePicker } from 'homeday-blocks';

export default {
  components: { HdDatePicker },
  data() {
    return {
      date: '2020-06-01',
    };
  },
};
</script>
```
