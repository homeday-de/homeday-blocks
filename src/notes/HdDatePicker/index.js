import defaultNotes from './default.md';
import disabledNotes from './disabled.md';
import disablingDatesNotes from './disablingDates.md';
import displayFormatNotes from './displayFormat.md';
import startWeekOnSundayNotes from './startWeekOnSunday.md';

export {
  defaultNotes,
  disabledNotes,
  disablingDatesNotes,
  displayFormatNotes,
  startWeekOnSundayNotes,
};
