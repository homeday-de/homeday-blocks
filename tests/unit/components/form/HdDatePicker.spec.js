import { wrapperFactoryBuilder } from 'tests/unit/helpers';

import HdDatePicker from '@/components/form/HdDatePicker.vue';

const wrapperBuilder = wrapperFactoryBuilder(HdDatePicker, {});

describe('HdDatePicker', () => {
  it('is rendered as expected', () => {
    const wrapper = wrapperBuilder();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
