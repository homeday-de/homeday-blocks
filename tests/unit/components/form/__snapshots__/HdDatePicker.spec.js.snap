// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdDatePicker is rendered as expected 1`] = `
<div class="vdpComponent vdpWithInput"><input type="text">
  <!---->
  <!---->
</div>
`;
